localstack-core/localstack/services/ec2/patches.py

```diff
@@ -133,9 +127,9 @@ def ec2_create_security_group(
 
         if custom_id:
             # Remove the security group from the default dict and add it back with the custom id
-            self.groups[vpc_id].pop(result.group_id)
+            self.groups[result.vpc_id].pop(result.group_id)
             result.group_id = result.id = custom_id
-            self.groups[vpc_id][custom_id] = result
```

Comment:
This could crash if VpcId was null which is the case for auto assigning to the Default Vpc